src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,156 @@
+ *                           and the left edge of the wrapped component.
+ * @param {Number} yOffset - The distance between the top edge of the window
+ *                           and the top edge of the wrapped component.
+ * @param {Number} componentWidth - The width of the wrapped component.
+ * @param {Number} componentHeight - The height of the wrapped component.
+ * @param {Number} tooltipWidth - The width of the tooltip itself.
+ * @param {Number} tooltipHeight - The height of the tooltip itself.
+ * @returns {Object}
+ */
+export default function getTooltipStyles(
+    interpolatedSize,
+    windowWidth,
+    xOffset,
+    yOffset,
+    componentWidth,
+    componentHeight,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    // If a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+    // we'll display it beneath its wrapped component rather than above it as usual.
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth);
+
+    return {
+        animationStyle: {
+            transform: [{
+                scale: interpolatedSize,
+            }],
+        },
+        tooltipWrapperStyle: {
+            position: 'absolute',
+            backgroundColor: themeColors.heading,
+            borderRadius: variables.componentBorderRadiusSmall,
+            ...spacing.pv1,
+            ...spacing.ph2,
+
+            // Shift the tooltip up by...
+            //   11 (font height)
+            // + 4 (top padding)
+            // + 4 (bottom padding)
+            // + 4 (upward shift of the pointer)
+            // + 3 (pointer height - 1)
+            // = 26
+            // OR
+            // Shift the tooltip down by...
+            // (component height) + (pointer height - 1)
+            top: shouldShowBelow ? componentHeight + 3 : -26,
+
+            // Shift the tooltip to the left by...
+            //   half the component's width
+            // + half the tooltip's width
+            // + any horizontal shift
+            left: (tooltipWidth / -2) + (componentWidth / 2) + horizontalShift,
+        },
+        tooltipTextStyle: {
+            color: themeColors.textReversed,
+            fontFamily: fontFamily.GTA,
+            fontSize: variables.fontSizeSmall,
+        },
+        pointerWrapperStyle: {
+            position: 'absolute',
+
+            // Shift the pointer up by its height
+            // OR
+            // Down by the component's height - its height
+            top: shouldShowBelow ? componentHeight - 4 : -4,
+
+            // Shift the pointer to the right (to the middle of the wrapped element)
+            left: (componentWidth / 2) - 4,
```

Comment:
Looks like this 4 has a relationship with the 4 in the `pointerStyle` so maybe it should use a variable.